### PR TITLE
Migrate `asset_handler_test` to null-safety

### PR DESCRIPTION
**Pull request by elliette**

### CODE REVIEW

File: dwds/lib/src/dwds_vm_client.dart [hunk 1]
```diff
@@ -55,7 +55,7 @@ class DwdsVmClient {
             '$request');
         return;
       }
-      requestController.sink.add(jsonDecode(request) as Map<String, Object>);
+      requestController.sink.add(Map<String, Object>.from(jsonDecode(request)));
```
Comment (annagrin) [inline dwds/lib/src/dwds_vm_client.dart:58]: I wonder if it is necessary to create a new object here (possible performance hit?)

File: dwds/lib/src/services/debug_service.dart [hunk 2]
```diff
@@ -53,7 +53,7 @@ void Function(WebSocketChannel, String) _createNewConnectionHandler(
             'Got value with unexpected type ${value.runtimeType} from web '
             'socket, expected a List<int> or String.');
       }
-      final request = jsonDecode(value as String) as Map<String, Object>;
+      final request = Map<String, Object>.from(jsonDecode(value));
```
Comment (annagrin) [inline dwds/lib/src/services/debug_service.dart:56]: Is the new object necessary here as well?

**Review by annagrin — APPROVED** [review]
Thanks Elliott! LGTM with a couple of questions an an optional ask.

File: dwds/test/fixtures/context.dart [hunk 3]
```diff
@@ -353,11 +354,12 @@ class TestContext {
           : 'http://localhost:$port/$basePath/$path';
 
       await _webDriver?.get(appUrl);
-      final tab = await (connection.getTab((t) => t.url == appUrl)
-          as FutureOr<ChromeTab>);
-      _tabConnection = await tab.connect();
-      await tabConnection.runtime.enable();
-      await tabConnection.debugger.enable();
+      final tab = await connection.getTab((t) => t.url == appUrl);
+      if (tab != null) {
```
[the remark this answers is not in the document]
Comment (elliette) [reply dwds/test/fixtures/context.dart:359]: Good point, done!

File: dwds/lib/src/dwds_vm_client.dart (hunk 1 above)
Comment (elliette) [reply dwds/lib/src/dwds_vm_client.dart:58]: I'm not sure of another way - `jsonDecode` returns `dynamic`, so using `as` to cast is throwing a type error (`as` should only be used "if and only if you are sure that the object is of that type": https://dart.dev/guides/language/language-tour#type-test-operators)

**Pull request merged**